integrations/helm/release/utils_test.go

```diff
@@ -0,0 +1,168 @@
+		},
+	},
+	"resources": map[string]interface{}{
+		"memory": "1Gi",
+	},
+}
+
+var collectNestedListParams = []ifv1.HelmChartParam{
+	{Name: "image.repository.name", Value: `[{"key":"aaa","value":"val1"},{"key":"bbb","value":"val2"}]`},
+	{Name: "resources.memory", Value: `[{"key":"ccc","value":"val3"},{"key":"ddd","value":"val4"}]`},
+}
+var collectExpectedNestedList = map[string]interface{}{
+	"image":     map[string]interface{}{"repository": map[string]interface{}{"name": []interface{}{map[string]interface{}{"key": "aaa", "value": "val1"}, map[string]interface{}{"key": "bbb", "value": "val2"}}}},
+	"resources": map[string]interface{}{"memory": []interface{}{map[string]interface{}{"key": "ccc", "value": "val3"}, map[string]interface{}{"key": "ddd", "value": "val4"}}}}
+
+var collectNestedMixtureParams = []ifv1.HelmChartParam{
+	{Name: "image.repository.name", Value: "bitnami/kibana"},
+	{Name: "resources.memory", Value: `[{"key":"aaa","value":"val"},{"key":"bbb","value":"val2"}]`},
+}
+var collectExpectedNestedMixture = map[string]interface{}{
+	"image": map[string]interface{}{
+		"repository": map[string]interface{}{
+			"name": "bitnami/kibana",
+		},
+	},
+	"resources": map[string]interface{}{"memory": []interface{}{map[string]interface{}{"key": "aaa", "value": "val"}, map[string]interface{}{"key": "bbb", "value": "val2"}}}}
+
+// fixtures for unwrap tests
+func TestMappifyValueOverride(t *testing.T) {
+	for _, c := range []mappifyTestCase{
+		{"simple parameter", "image", "bitnami/kibana:master-1", mappifyExpectedSimple},
+		{"nested parameter: 2 levels", "image.repository", "bitnami/kibana:master-1", mappifyExpectedNested2},
+		{"nested parameter: 3 levels", "image.repository.name", "bitnami/kibana:master-1", mappifyExpectedNested3},
+	} {
+		testMappify(t, c)
+	}
+}
+
+func TestCollectValues(t *testing.T) {
+	for _, c := range []collectTestCase{
+		{"nil params", nil, map[string]interface{}{}},
+		{"nil params", []ifv1.HelmChartParam{}, map[string]interface{}{}},
+		{"simple parameters, simple value", collectSimpleSimpleParams, collectExpectedSimpleSimple},
+		{"simple parameters: list value", collectSimpleListParams, collectExpectedSimpleList},
+		{"nested parameters: simple values", collectNestedSimpleParams, collectExpectedNestedSimple},
+		{"nested parameters: list values", collectNestedListParams, collectExpectedNestedList},
+		{"nested parameters: simple and list values", collectNestedMixtureParams, collectExpectedNestedMixture},
+	} {
+		testCollect(t, c)
+	}
+}
+
+func TestUnwrap(t *testing.T) {
+	for _, c := range []unwrapTestCase{
+		{"list with simple values", `["AAA","BBB","CCC"]`, []interface{}{"AAA", "BBB", "CCC"}},
+		{"list with maps", `[{"key":"DDD","value":"ddd"},{"key":"EEE","value":"eee"}]`, []interface{}{map[string]interface{}{"key": "DDD", "value": "ddd"}, map[string]interface{}{"key": "EEE", "value": "eee"}}},
+	} {
+		testUnwrap(t, c)
+	}
+}
+
+func testMappify(t *testing.T, c mappifyTestCase) {
```

Comment:
would it be easier to read tests if this code is in the test function? this code is not used elsewhere.